### PR TITLE
Clarify & expand service decoration description

### PR DESCRIPTION
**Pull request by gnito-org**
I also added an App\Mailer namespace to all code examples.

### CODE REVIEW

**Comment by gnito-org** [comment]
@javiereguiluz Pinging for a review when you have time. :pray:  :smile:

File: service_container/service_decoration.rst [hunk 1]
```diff
@@ -246,6 +246,213 @@ automatically changed to ``'.inner'``):
...
+
+.. tip::
+
+    This first method is the recommended way to decorate a service.
+
+    However, it only works when the decorated class implements an injectable interface.
+
+Assume the following for the decorated class::
+
+    // Mailer/Mailer.php
```
Comment (OskarStark) [inline service_container/service_decoration.rst]:
```suggestion
    // src/Mailer/Mailer.php
```

Same for the other paths in the code blocks

**Review by OskarStark — COMMENTED** [review]
Can you explain why we need the sub Mailer Namespace?

**Comment by OskarStark** [comment]
@nicolas-grekas would you have some time to review this PR ? Thanks

**Comment by gnito-org** [comment]
> Can you explain why we need the sub Mailer Namespace?

@OskarStark Thank you for the review! I added the sub Mailer namespace because that's how it would normally be coded (best practice). We would not put the ``Mailer.php`` class in the root of the ``src`` folder.

**Comment by gnito-org** [comment]
I unintentionally included a [commit](https://github.com/symfony/symfony-docs/pull/16161/commits/09a632b3e7f0487f0685b5cd78b361c5826c79b9), what was supposed to be a separate PR, in this PR.

Can it remain in this PR? Honestly, I do not know how to remove it.

The commit was in reponse to https://github.com/symfony/symfony-docs/issues/16149.

**Comment by gnito-org** [comment]
I'm closing this PR and redoing it because I forgot to create separate fix branches. That's basically preventing me from creating separate new PR.

**Pull request closed without merging**